### PR TITLE
Update Dependencies

### PR DESCRIPTION
**Pull request by chrhicks**
Update Development Dependencies:
* Sync yarn lockfile
* Disable new `react/no-typos` eslint rule introduced by `@flowio/eslint-config-flowio`
* Add `enzyme-adapter-react-15` version 1.0.1
* Update `@flowio/eslint-config-flowio` to version 11.0.1 from 10.0.0
* Update `babel-eslint` to version 8.0.1 from 8.0.0
* Update `babel-plugin-istanbul` to version 4.1.5 from 4.1.4
* Update `enzyme` to version 3.1.0 from 2.9.1
* Update `eslint` to version 4.8.0 from 4.6.0
* Update `eslint-plugin-jsx-a11y` to version 6.0.2 from 5.1.1
* Update `eslint-plugin-react` to version 7.4.0 from 7.3.0
* Update `jsdom` to version 11.3.0 from 11.2.0
* Update `mocha` to version 4.0.1 from 3.5.3
* Update `prop-types` to version 15.6.0 from 15.5.10
* Update `sinon` to version 4.0.1 from 3.0.0
* Update `sinon-chai` to version 2.14.0 from 2.13.0

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@flowio/redux-fetch",
-  "version": "0.14.1",
```
Comment (ctumolosus) [inline package.json]: No need to bump package version. Generally, I run `npm version` before publishing the package. This has the potential of assigning the wrong tag.

File: test/specs/FetchRootContainer.jsx [hunk 2]
```diff
@@ -14,64 +14,65 @@ describe('FetchRootContainer', () => {
...
 
     expect(onFetchRouteData).to.not.have.been.called;
   });
 
   it('should call onFetchRouteData when data requirements are not met on mount', () => {
     const routerState = createMockRouterState();
     const onFetchRouteData = sinon.stub();
-
-    mount(
+    const jsx = (
```
Comment (ctumolosus) [inline test/specs/FetchRootContainer.jsx]: I'm more comfortable adding this to `.eslintrc`:

```json
{
  "function-paren-newline": ["error", "consistent"]
}
```

File: examples/twitter/server/controllers/renderer.jsx [hunk 3]
```diff
@@ -38,13 +38,14 @@ function renderNotFound() {
 }
 
 function renderToString(store, renderProps) {
-  return ReactDOMServer.renderToString(
+  const jsx = (
```
Comment (chrhicks) [inline examples/twitter/server/controllers/renderer.jsx]: fixed already, commit incoming.

Comment (ctumolosus) [inline examples/twitter/server/controllers/renderer.jsx]: Can restore this back too.

**Pull request merged**